common/src/main/java/us/ilite/common/lib/control/PIDController.java

```diff
@@ -0,0 +1,331 @@
+     *
+     * @return integral coefficient
+     */
+    public double getI() {
+        return m_I;
+    }
+
+    /**
+     * Get the Differential coefficient
+     *
+     * @return differential coefficient
+     */
+    public double getD() {
+        return m_D;
+    }
+
+    /**
+     * Get the Feed forward coefficient
+     *
+     * @return feed forward coefficient
+     */
+    public double getF() {
+        return m_F;
+    }
+
+    /**
+     * Return the current PID result This is always centered on zero and constrained the the max and min outs
+     *
+     * @return the latest calculated output
+     */
+    public double get() {
+        return m_result;
+    }
+
+    /**
+     * Set the PID controller to consider the input to be continuous, Rather then using the max and min in as
+     * constraints, it considers them to be the same point and automatically calculates the shortest route to the
+     * setpoint.
+     *
+     * @param continuous
+     *            Set to true turns on continuous, false turns off continuous
+     */
+    public void setContinuous(boolean continuous) {
+        m_continuous = continuous;
+    }
+
+    public void setDeadband(double deadband) {
+        m_deadband = deadband;
+    }
+
+    /**
+     * Set the PID controller to consider the input to be continuous, Rather then using the max and min in as
+     * constraints, it considers them to be the same point and automatically calculates the shortest route to the
+     * setpoint.
+     */
+    public void setContinuous() {
+        this.setContinuous(true);
+    }
+
+    /**
+     * Sets the maximum and minimum values expected from the input.
+     *
+     * @param minimumInput
+     *            the minimum value expected from the input
+     * @param maximumInput
+     *            the maximum value expected from the output
+     */
+    public void setInputRange(double minimumInput, double maximumInput) {
+        if (minimumInput > maximumInput) {
+            System.out.println("Lower bound is greater than upper bound");
+        }
+        m_minimumInput = minimumInput;
+        m_maximumInput = maximumInput;
+        setSetpoint(m_setpoint);
+    }
+
+    /**
+     * Sets the minimum and maximum values to write.
+     *
+     * @param minimumOutput
+     *            the minimum value to write to the output
+     * @param maximumOutput
+     *            the maximum value to write to the output
+     */
+    public void setOutputRange(double minimumOutput, double maximumOutput) {
+        if (minimumOutput > maximumOutput) {
+            System.out.println("Lower bound is greater than upper bound");
+        }
+        m_minimumOutput = minimumOutput;
+        m_maximumOutput = maximumOutput;
+    }
+
+    /**
+     * Set the setpoint for the PID controller
+     *
+     * @param setpoint
+     *            the desired setpoint
+     */
+    public void setSetpoint(double setpoint) {
+        if (m_maximumInput > m_minimumInput) {
```

Comment:
consider using Math.max() and Math.min() instead of complex if() blocks for numbers

Reply:
Could I also use Util.limit() (2019 version of clamp from last year)?

Reply:
Probably (having not seen that method)